src/Formats/FormatFactory.cpp

```diff
@@ -777,13 +786,13 @@ String FormatFactory::getAdditionalInfoForSchemaCache(const String & name, Conte
 
 bool FormatFactory::isInputFormat(const String & name) const
 {
-    auto it = dict.find(name);
+    auto it = dict.find(getOriginalFormatNameIfExists(name));
```

Comment:
If all formats are case insensitive, we should just store their lowercase value in the hash table and look for lowercase here. No need of `getOriginalFormatNameIfExists`

Reply:
Now, it's just converted to lowercase.

Reply:
1. I found that using lowercase directly causes the name column in the `system.forms` table to be all lowercase, in `StorageSystemFormats::fillData()`
2. Many of the functional tests use name for queries (such as, `select number_of_rows from system.schema_inference_cache where format='Npy'` in 02908_Npy_files_caching), so it is necessary to use original format name